app/scripts/modules/kubernetes/src/validation/applicationName.validator.js

```diff
@@ -10,11 +10,16 @@ module.exports = angular
   ])
   .factory('kubernetesApplicationNameValidator', function () {
 
-    function validateSpecialCharacters(name, errors) {
+    function validateSpecialCharacters(name, warnings, errors) {
       let pattern = /^([a-zA-Z][a-zA-Z0-9]*)?$/;
       if (!pattern.test(name)) {
```

Comment:
What purpose does the first pattern test serve here?  It looks like we only need the new pattern that includes the dash - is that right?

Reply:
To distinguish between showing an error ("foo&", "foo*bar") and a warning ("foo-", "foo-bar"). I'm happy to add comments explaining the logic. Renaming the patterns would probably help a bit as well.

Reply:
Ahh I understand now - thanks, makes sense.